src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -440,12 +442,50 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 				if (this._isCellStyledLikeGhostText(cell)) {
 					ghostTextIndex = cursorIndex + potentialGhostIndexOffset;
 					break;
+				} else if (!this._cellStylesMatch(cursorCell, cell)) {
+					const ghostTextCell = cell;
+					// Verify that all chars for the rest of the line match this
+					// ghost text style
+					while (x < line.length) {
+						const cell = line.getCell(x++);
+						if (!cell || cell.getCode() === 0) {
+							break;
+						}
+
+						if (!this._cellStylesMatch(ghostTextCell, cell)) {
+							ghostTextIndex = -1;
+							break;
```

Comment:
Should this return -1 instead? I think the `= -1` is a no-op because it gets overwritten down below